src/main/groovy/life/qbic/db/MariaDBManager.groovy

```diff
@@ -500,7 +498,47 @@ class MariaDBManager implements IQueryService {
     return res
   }
 
-  //TODO JavaDoc
+  @Override
+  void sampleChanged(String sampleCode, Status sampleStatus) {
+    Connection connection = Objects.requireNonNull(dataSource.getConnection(),
+            "Connection must not be null.")
+    Sql sql = new Sql(connection)
+    try {
+      logSampleChange(sampleCode, sampleStatus, sql)
+    } catch (Exception unexpected) {
+      log.error("An unexpected error occured: $unexpected.message")
+      log.debug("An unexpected error occured: $unexpected.message", unexpected)
+    } finally {
+      sql.close()
+    }
+  }
+
+  /**
+   * Writes a sample status change to the notification table.
+   * @param sampleCode the the sample code of the changed sample
+   * @param sampleStatus the new value of the sample status
+   * @param sql The sql connection facade to be used
+   * @see #sampleChanged
+   */
+  private static void logSampleChange(String sampleCode, Status sampleStatus, Sql sql) {
+    String query = "INSERT INTO notification (`sample_code`, `arrival_time`, `sample_status`) " +
+            "VALUES(?, CURRENT_TIMESTAMP, ?);"
```

Comment:
where does "CURRENT_TIMESTAMP" come from?

Reply:
This is SQL and provides the current time :)

Reply:
nice